queue/config.go

```diff
@@ -25,18 +29,118 @@ func DefaultConfig() Config {
 			Events:         4096,
 			FlushMinEvents: 2048,
 			FlushTimeout:   1 * time.Second,
-		}, //memqueue should have a DefaultSettings()
-		DiskSettings: nil,
+		}, // memqueue should have a DefaultSettings()
+		DiskConfig: nil,
 	}
 }
 
 func (c *Config) Validate() error {
-	if c.MemSettings == nil && c.DiskSettings == nil {
+	if c.MemSettings == nil && c.DiskConfig == nil {
 		return fmt.Errorf("memory or disk queue settings must be supplied")
 	}
 	return nil
 }
 
 func (c Config) useDiskQueue() bool {
-	return c.DiskSettings != nil
+	return c.DiskConfig != nil
+}
+
+type DiskConfig struct {
```

Comment:
Can you update the reference configuration with these parameters as well? https://github.com/elastic/elastic-agent-shipper/blob/cb5cf808608678de86f85d353ab7fb1c41260465/elastic-agent-shipper.yml#L79

For now we have to manually keep this in sync, until we can come up with a better way.

It is likely also worth nothing that when the shipper is started by the agent that configuration isn't written to disk, such that `EncryptionPassword` is never viewable in plaintext.